### PR TITLE
switch GPU CI to buildkite

### PR DESCRIPTION
**Pull request by simonbyrne**

### CODE REVIEW

File: src/cuda.jl [hunk 1]
```diff
@@ -5,6 +5,7 @@ function Base.cconvert(::Type{MPIPtr}, buf::CUDA.CuArray{T}) where T
 end
 
 function Base.unsafe_convert(::Type{MPIPtr}, X::CUDA.CuArray{T}) where T
+    synchronize(CuDefaultStream())
```
Comment (simonbyrne) [inline src/cuda.jl:8]: This is one possible solution to #443, but is probably overkill.

File: src/cuda.jl [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import .CUDA
 
 function Base.cconvert(::Type{MPIPtr}, buf::CUDA.CuArray{T}) where T
+    CUDA.synchronize(CUDA.CuDefaultStream())
```
Comment (simonbyrne) [inline src/cuda.jl:4]: @vchuravy What do you think?

**Comment by simonbyrne** [comment]
The decision here was to require users to manually synchronise. We will need to update docs and the tests.

File: src/cuda.jl (hunk 2 above)
Comment (vchuravy) [reply src/cuda.jl:4]: Why do we need this again?

cc: @maleadt

Comment (maleadt) [reply src/cuda.jl:4]: https://github.com/JuliaParallel/MPI.jl/issues/443

**Comment by vchuravy** [comment]
No part of #552

**Pull request closed without merging**